qiskit/aqua/operators/primitive_ops/pauli_sum_op.py

```diff
@@ -185,7 +185,9 @@ def compose(
 
     def to_matrix(self, massive: bool = False) -> np.ndarray:
         OperatorBase._check_massive("to_matrix", True, self.num_qubits, massive)
-        return self.primitive.to_matrix() * self.coeff  # type: ignore
+        if isinstance(self.coeff, ParameterExpression):
+            return (self.primitive.to_matrix(sparse=True)).toarray() * self.coeff  # type: ignore
```

Comment:
Does `return (self.primitive.to_matrix(sparse=False)) * self.coeff` work?

Reply:
Yes, but this is better from the perspective of performance.
(Pauli.to_matrix() also uses spmatrix and toarray().)

Reply:
I see. Because the matrix is very sparse, it is beneficial to take the sum in a sparse way and convert it into a dense matrix later.